packages/table/examples/tableLoadingExample.tsx

```diff
@@ -0,0 +1,113 @@
+    public state: ITableLoadingExampleState = {
+        tableLoadingOptions: [ TableLoadingOption.CELLS ],
+    };
+
+    protected className = "docs-table-loading-example";
+
+    protected renderExample() {
+        return (
+            <Table
+                loadingOptions={this.state.tableLoadingOptions}
+                numRows={fuelConsumptionData.length}
+                renderRowHeader={this.renderRowHeader}
+            >
+                {this.renderColumns()}
+            </Table>
+        );
+    }
+
+    protected renderOptions() {
+        /* tslint:disable:jsx-no-lambda */
+        return [
+            [
+                <Switch
+                    checked={this.hasLoadingOption(TableLoadingOption.CELLS)}
+                    key="cells"
+                    label="Cell Loading"
+                    onChange={() => this.toggleLoadingOption(TableLoadingOption.CELLS)}
+                />,
+                <Switch
+                    checked={this.hasLoadingOption(TableLoadingOption.COLUMN_HEADERS)}
+                    key="column-headers"
+                    label="Column Header Loading"
+                    onChange={() => this.toggleLoadingOption(TableLoadingOption.COLUMN_HEADERS)}
+                />,
+                <Switch
+                    checked={this.hasLoadingOption(TableLoadingOption.ROW_HEADERS)}
+                    key="row-headers"
+                    label="Row Header Loading"
+                    onChange={() => this.toggleLoadingOption(TableLoadingOption.ROW_HEADERS)}
+                />,
+            ],
+        ];
+        /* tslint:enable:jsx-no-lambda */
+    }
+
+    private renderColumns() {
+        const columns: JSX.Element[] = [];
+        const renderCell = (rowIndex: number, columnIndex: number) => {
+            return <Cell>{fuelConsumptionData[rowIndex][columnIndex + 2000]}</Cell>;
+        };
+
+        Object.keys(fuelConsumptionData[0]).forEach((key) => {
+            if (key !== "country") {
+                columns.push(
+                    <Column
+                        name={key}
+                        key={key}
+                        renderCell={renderCell}
+                    />,
+                );
+            }
+        });
+
+        return columns;
+    }
+
+    private renderRowHeader = (rowIndex: number) => {
+        return <RowHeaderCell name={fuelConsumptionData[rowIndex].country} style={{ width: 150 }}/>;
+    }
+
+    private hasLoadingOption = (loadingOption: RegionCardinality) => {
+        return this.state.tableLoadingOptions.indexOf(loadingOption) !== -1;
+    }
+
+    private toggleLoadingOption = (loadingOption: RegionCardinality) => {
+        const { tableLoadingOptions } = this.state;
+        const optionIndex = tableLoadingOptions.indexOf(loadingOption);
+        if (optionIndex !== -1) {
+            tableLoadingOptions.splice(optionIndex, 1);
+        } else {
+            tableLoadingOptions.push(loadingOption);
```

Comment:
it is _likely_ that mutating the actual array prevents React from being able to diff the old and new states _cuz you changed the original value_. try creating new arrays each time like so:

```js
this.setState({ tableLoadingOptions: [...this.state.tableLoadingOptions, loadingOption] })
```